pyperformance/data-files/benchmarks/bm_ctypes/run_benchmark.py

```diff
@@ -0,0 +1,42 @@
+"""
+Test the function call overhead of ctypes.
+"""
+import pyperf
+
+
+import ctypes
+import importlib.util
+
+
+spec = importlib.util.find_spec("bm_ctypes.cmodule")
+if spec is None:
+    raise ImportError("Can't find bm_ctypes.cmodule shared object file")
+ext = ctypes.cdll.LoadLibrary(spec.origin)
+
+
+def benchmark(n):
+    void_foo_void = ext.void_foo_void
+    int_foo_int = ext.int_foo_int
+    void_foo_int = ext.void_foo_int
+    void_foo_int_int = ext.void_foo_int_int
+    void_foo_int_int_int = ext.void_foo_int_int_int
+    void_foo_int_int_int_int = ext.void_foo_int_int_int_int
+    void_foo_constchar = ext.void_foo_constchar
```

Comment:
For *micro* benchmarks using `runner.bench_func()` we want to keep the function body as tight as possible to the code we want to measure.   Depending on how fast each loop iteration is and what `n` is, it may make sense to move some of this out of the function:

```suggestion
_void_foo_void = ext.void_foo_void
_int_foo_int = ext.int_foo_int
_void_foo_int = ext.void_foo_int
_void_foo_int_int = ext.void_foo_int_int
_void_foo_int_int_int = ext.void_foo_int_int_int
_void_foo_int_int_int_int = ext.void_foo_int_int_int_int
_void_foo_constchar = ext.void_foo_constchar


def benchmark(n):
    void_foo_void = _void_foo_void
    int_foo_int = _int_foo_int
    void_foo_int = _void_foo_int
    void_foo_int_int = _void_foo_int_int
    void_foo_int_int_int = _void_foo_int_int_int
    void_foo_int_int_int_int = _void_foo_int_int_int_int
    void_foo_constchar = _void_foo_constchar
```
or:
```suggestion

def benchmark(n, *,
                         void_foo_void =ext.void_foo_void,
                         int_foo_int=xt.int_foo_int,
                         void_foo_int=ext.void_foo_int,
                         void_foo_int_int=ext.void_foo_int_int,
                         void_foo_int_int_int=ext.void_foo_int_int_int,
                         void_foo_int_int_int_int=ext.void_foo_int_int_int_int,
                         void_foo_constchar=ext.void_foo_constchar,
                         ):
```

But it's only worth it if ctypes is really fast.

Then again, the point is moot if you use `runner.bench_time_func()` instead of `runner.bench_func()`.